packages/grafana-data/src/utils/datasource.ts

```diff
@@ -22,8 +22,8 @@ export function getDataSourceRef(ds: DataSourceInstanceSettings): DataSourceRef
  *
  * @public
  */
-export function isDataSourceRef(ref: DataSourceRef | string | null): ref is DataSourceRef {
-  return typeof ref === 'object' && (typeof ref?.uid === 'string' || typeof ref?.uid === 'undefined');
+export function isDataSourceRef(ref: DataSourceRef | string | null | undefined): ref is DataSourceRef {
+  return typeof ref === 'object' && typeof ref?.uid === 'string';
```

Comment:
Although Levitate [highlighted it](https://github.com/grafana/grafana/runs/5232929151?check_suite_focus=true#step:8:28), I wouldn't consider it being a breaking change, so I think we should ignore the `breaking-change` notice.